met-web/src/components/engagement/admin/view/AuthoringTab.tsx

```diff
@@ -0,0 +1,197 @@
+    const optionalSectionTitles = ['View Results', 'Subscribe', 'More Engagements'];
+    const feedbackTitles = ['Survey', '3rd Party Feedback Method Link'];
+    const defaultAuthoringValue: AuthoringValue = {
+        title: '',
+        link: '#',
+        required: false,
+        completed: false,
+    };
+    const getAuthoringValues = (defaultValues: AuthoringValue, titles: string[], required: boolean): AuthoringValue[] =>
+        titles.map((title) => ({ ...defaultValues, title: title, required: required }));
+    const mandatorySectionValues = getAuthoringValues(defaultAuthoringValue, mandatorySectionTitles, true);
+    const optionalSectionValues = getAuthoringValues(defaultAuthoringValue, optionalSectionTitles, false);
+    const defaultSectionValues = [...mandatorySectionValues, ...optionalSectionValues];
+    const defaultFeedbackMethods = getAuthoringValues(defaultAuthoringValue, feedbackTitles, true);
+
+    // Set useStates. When data is imported, it will be set with setSectionValues and setFeedbackMethods.
+    const [sectionValues, setSectionValues] = useState(defaultSectionValues);
+    const [feedbackMethods, setFeedbackMethods] = useState(defaultFeedbackMethods);
+    const [requiredSectionsCompleted, setRequiredSectionsCompleted] = useState(false);
+    const [feedbackCompleted, setFeedbackCompleted] = useState(false);
+
+    // Define styles
+    const systemMessageStyles = {
+        marginBottom: '1.5rem',
+    };
+    const metHeaderStyles = {
+        marginBottom: '1.5rem',
+        fontSize: '1.2rem',
+    };
+    const metLabelStyles = {
+        textTransform: 'uppercase',
+        marginBottom: '1.1rem',
+        fontSize: '0.9rem',
+    };
+    const anchorContainerStyles = {
+        margin: '0 0 2.5rem 0',
+        padding: '0',
+    };
+
+    // Listen to the sectionValues useState and update the boolean value that controls the presence of the "sections" incomplete message.
+    useEffect(() => {
+        if (true !== requiredSectionsCompleted && allRequiredItemsComplete(sectionValues)) {
+            setRequiredSectionsCompleted(true);
+        } else if (false !== requiredSectionsCompleted) {
+            setRequiredSectionsCompleted(false);
+        }
+    }, [sectionValues]);
+
+    // Listen to the feedbackMethods useState and update the boolean value that controls the presence of the "feedback" incomplete message.
+    useEffect(() => {
+        if (true !== feedbackCompleted && allRequiredItemsComplete(feedbackMethods)) {
+            setFeedbackCompleted(true);
+        } else if (false !== feedbackCompleted) {
+            setRequiredSectionsCompleted(false);
+        }
+    }, [feedbackMethods]);
+
+    // Check if all required items are completed.
+    const allRequiredItemsComplete = (values: AuthoringValue[]) => {
+        const itemChecklist = values.map((value) => {
+            if (undefined === value.required || undefined === value.completed) {
```

Comment:
Right, you wouldn't have an undefined value from the default data. It's when you start introducing real data that this becomes important.

Reply:
I'm okay with proceeding with this code but I thought I'd add a note that with Typescript, we often don't need to account for `undefined` popping up. We have a lot more type control which allows us to not have to worry about checking for `undefined` in case things break.